src/js/utils/crypto/index.ts

```diff
@@ -0,0 +1,137 @@
+function arrayBufferToBase64(buffer: ArrayBuffer): string {
+  let binary = "";
+  let bytes = new Uint8Array(buffer);
+  for (var i = 0; i < bytes.byteLength; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
+  return window.btoa(binary);
+}
+
+function getCryptoApi(): SubtleCrypto {
+  const cryptoLib = window.crypto || (window as any).msCrypto;
+  return cryptoLib.subtle || (cryptoLib as any).webkitSubtle;
```

Comment:
i'd love comments about why we need `msCrypto` and `webkitSubtle`. to me it looks like "IE <= 11" and "safari".

Reply:
The reason why is because it has been supported without prefix for the past 2 major versions

Reply:
These two browser prefixed different objects in the crypto API. IE 11 prefixed crypto, but safari prefixed subtle!

Reply:
thanks! 👍